docs/DG_Diagrams/LogicComponent.puml

```diff
@@ -0,0 +1,48 @@
+@startuml
+
+allowmixing
+Rectangle Logic {
+Rectangle commands {
+Rectangle commons {
+abstract class Command
+Command <|-left ExitCommand
+Command <|- HelpCommand
+}
+Rectangle buscommand {
+Command <|-- AllBusCommand
+Command <|-left BusCommand
+Command <|-- ListStopsCommand
+Command <|-- ResetSearchFreqCommand
+Command <|-- RouteCommand
+Command <|-- RouteMapCommand
+}
+Rectangle dinecommand {
+Command <|-left DineCommand
+Command <|-left DineInfoCommand
+}
+Rectangle favcommand {
+Command <|--up AddFavCommand
+Command <|--up ClearFavCommand
+Command <|--up DeleteFavCommand
+Command <|--up DescFavCommand
+Command <|-- ExecFavCommand
+Command <|-- ListFavCommand
+}
+}
+Rectangle parser {
+Parser <|-- RouteParser
+RouteParser <-down RouteCommand
+Parser <|-- DescFavParser
+DescFavParser <- DescFavCommand
+Parser <|-- ExecFavParser
+ExecFavParser <- ExecFavCommand
+}
+class SimilarityCheck
+
+SimilarityCheck <..down RouteCommand
+SimilarityCheck <..down BusCommand
+}
+
+Parser --left> commands
+
```

Comment:
![Screenshot_1](https://user-images.githubusercontent.com/59549150/97513204-95a88380-19c6-11eb-84e1-26ad6be5ed7f.png)

The arrows looks a little messy as compared to the other diagrams. Maybe try to make the arrows more straight?
Can also consider to use XCommand to represent the different commands in each of the boxes, to reduce repetition in the diagram, and explain its representation in the explanation for the diagram.